doc/tutorials/menu/index.rst

```diff
@@ -0,0 +1,289 @@
+.. include:: <isonum.txt>
+
+.. _menu_tutorial:
+
+
+Making a Menu with Arcade's GUI
+===============================
+
+.. image:: menu.gif
+    :width: 80%
+
+This tutorial shows how to use most of arcade's gui's widgets.
+
+Step 1: Open a Window
+---------------------
+
+.. image:: menu_01.png
+    :width: 50%
+
+First, let's start a blank window with a view.
+
+.. literalinclude:: menu_01.py
+    :caption: Opening a Window
+    :linenos:
+
+Step 2: Switching to Menu View
+-------------------------------
+
+.. image:: menu_02.png
+    :width: 50%
+
+For this section we will switch the current view of the window to the menu view.
+
+
+Imports
+~~~~~~~
+
+First we will import the arcade gui:
+
+.. literalinclude:: menu_02.py
+    :caption: Importing arcade.gui
+    :lines: 4-5
+    :emphasize-lines: 2
+
+Modify the MainView
+~~~~~~~~~~~~~~~~~~~~
+
+We are going to add a button to change the view. For drawing a button we would need a ``UIManager``.
+
+.. literalinclude:: menu_02.py
+    :caption: Intialising the Manager
+    :lines: 16-19
+    :emphasize-lines: 4
+
+After initialising the manager we need to enable it when the view is shown and disable it when the view is hiddien.
+
+.. literalinclude:: menu_02.py
+    :caption: Enabling the Manager
+    :pyobject: MainView.on_show_view
+    :emphasize-lines: 5-6
+
+.. literalinclude:: menu_02.py
+    :caption: Disabling the Manager
+    :pyobject: MainView.on_hide_view
+
+We also need to draw the childrens of the menu in ``on_draw``.
+
+ .. literalinclude:: menu_02.py
+    :caption: Drawing Children's of the Manager
+    :pyobject: MainView.on_draw
+    :emphasize-lines: 6-7
+
+Now we have successfully setup the manager, only thing left it to add the button. We are using ``UIAnchorLayout`` to position the button. We also setup a function which is called when the button is clicked.
```

Comment:
Could we keep lines to about 80 characters? I know this isn't standard everywhere but I really, really, like coding w/o word wrap and no line breaks makes this stuff hard to read.

Reply:
Updated in https://github.com/pythonarcade/arcade/pull/1608/commits/a93bb6e03fedd180accf50b4ab101e7caf9dac29